contrib/platform/src/com/sun/jna/platform/mac/IOKit.java

```diff
@@ -0,0 +1,546 @@
+import com.sun.jna.platform.mac.CoreFoundation.CFMutableDictionaryRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFStringRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFTypeRef;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+import com.sun.jna.ptr.PointerByReference;
+
+/**
+ * The I/O Kit framework implements non-kernel access to I/O Kit objects
+ * (drivers and nubs) through the device-interface mechanism.
+ */
+public interface IOKit extends Library {
+
+    IOKit INSTANCE = Native.load("IOKit", IOKit.class);
+
+    int kIORegistryIterateRecursively = 0x00000001;
+    int kIORegistryIterateParents = 0x00000002;
+
+    double kIOPSTimeRemainingUnlimited = -2.0;
+    double kIOPSTimeRemainingUnknown = -1.0;
+
+    MachPort MACH_PORT_NULL = new MachPort();
+
+    /**
+     * IOKitLib implements non-kernel task access to common IOKit object types -
+     * IORegistryEntry, IOService, IOIterator etc. These functions are generic -
+     * families may provide API that is more specific.
+     * <p>
+     * IOKitLib represents IOKit objects outside the kernel with the types
+     * io_object_t, io_registry_entry_t, io_service_t, and io_connect_t. Function
+     * names usually begin with the type of object they are compatible with - e.g.,
+     * IOObjectRelease can be used with any io_object_t. Inside the kernel, the c++
+     * class hierarchy allows the subclasses of each object type to receive the same
+     * requests from user level clients, for example in the kernel, IOService is a
+     * subclass of IORegistryEntry, which means any of the IORegistryEntryXXX
+     * functions in IOKitLib may be used with io_service_t's as well as
+     * io_registry_t's. There are functions available to introspect the class of the
+     * kernel object which any io_object_t et al. represents. IOKit objects returned
+     * by all functions should be released with IOObjectRelease.
+     */
+    class IOObject extends PointerType {
+        public IOObject() {
+            super();
+        }
+
+        public IOObject(Pointer p) {
+            super(p);
+        }
+
+        /**
+         * Convenience method for {@link IOKit#IOObjectRelease} on this object.
+         *
+         * @return 0 if successful, otherwise a {@code kern_return_t} error code.
+         */
+        public int release() {
+            return INSTANCE.IOObjectRelease(this);
+        }
+    }
+
+    /**
+     * Communication between tasks is an important element of the Mach philosophy.
+     * Mach supports a client/server system structure in which tasks (clients)
+     * access services by making requests of other tasks (servers) via messages sent
+     * over a communication channel.
+     * <p>
+     * The endpoints of these communication channels in Mach are called ports, while
+     * port rights denote permission to use the channel.
+     */
+    class MachPort extends IOObject {
```

Comment:
Understood.  I was aiming for convenience, since I only really need this type for the master port (and even that can be omitted and use `null` most places!) and I know that the IOObjectRelease calls the appropriate mach function under the hood... but I can already hear you telling me "don't rely on implementation details". :)  

The Mach Port type is in the Kernel framework so it should be moved to the SystemB class (along with mapping the deallocate call).   I'll work on that this weekend.

Reply:
OK, fun times.   The `deallocate()` call requires an integer argument, which either requires jumping through hoops with pointer `peer` values, or revisiting this entire mapping.  And I finally figured out some of the confusion.   The same type (e.g., `mach_port_t`) represents a pointer on the kernel side, but an unsigned integer (pointer-sized) on the user-space side.

Which explains a lot of my earlier confusion and also matches with my observational data about these all being 32-bit values.  They aren't really pointers (even the "null pointer" is pointed to a `MACH_PORT_NULL` constant = 0.

So I think I'm going to map all these "port"-based-types (mach ports, and io_object_t) as `NativeLong`.

Reply:
As an example of the above, on the kernel side (C++) the `CFMachPort` type is `mach_port_t` and eventually points to a real structure with multiple members.  On the userspace side (C headers) the same `mach_port_t` points to an empty structure, basically just to get it a "pointer size integer" type, but not actually point to anything.